### ISSUE TITLE
Knowls in display mathematics

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#930]
See two knowls in an `md` at the end of 7.4 in

http://mathbook.pugetsound.edu/examples/sample-article/html/section-7.html

"Corollary" knowl behaves, "FTC" acts like MathJax collapsible math for me in Firefox.  This is the `\knowl{}` macro from http://aimath.org/mathbook/mathjaxknowl.js.

**Comment by rbeezer** [comment]
This seems to have fixed itself.  At least the offending example now behaves, and I've not seen other examples.

**Comment by Alex-Jordan** [comment]
When I turn on collapsible math, I can click in other math on that page and it collapses. When I click in the one with links, it's like collapsible math is disabled for the whole thing. Which is fine. Maybe that is how they (MathJax) addressed the issue. 

Related, if the zoom trigger is "click", then that overrules the link. Clicking zooms, not follows the link or opens the knowl. Just taking note of this.

**Comment by davidfarmer** [comment]
> When I turn on collapsible math, I can click in other math on that page and it collapses. When I click in
> the one with links, it's like collapsible math is disabled for the whole thing. Which is fine. Maybe that is
> how they (MathJax) addressed the issue.
> 
> Related, if the zoom trigger is "click", then that overrules the link. Clicking zooms, not follows the link
> or opens the knowl. Just taking note of this.

Maybe both of those could be put into a new issue?  We control the
extension that lets knowls work in display math.

**Comment by Alex-Jordan** [comment]
I just revisited the page, and I don't think the issue is resolved. Last night I did not understand that the Corollary and FTC link behaved differently. Now I am seeing the FTC link behave like collapsible math. (While clicking on the math does nothing). When this appeared resolved, did you have collapsible math turned on in the accessibility submenu of MathJax?

**Comment by rbeezer** [comment]
I've not investigated, but will now re-open this.

**Comment by davidfarmer** [comment]
I confirm that the FTC knowl acts like collapsible math when Collapsible Math is turned on.

Also, with Collapsible Math, the final equation on the page (the one with the star star) does
not get typeset by MathJax.

**Comment by rbeezer** [comment]
Do we think the "special" JS for this needs help or do we think that MathJax needs to get involved?

**Comment by rbeezer** [comment]
Very rough testing indicates MathJax 3 will solve this problem.  The two knowls mentioned here both behave using the experiment described on #1039.

**Comment by rbeezer** [comment]
I added an "upstream" tag so we know to wait on MJ3.

**Comment by rbeezer** [comment]
Added a third `xref` within Section 7.4 of the sample article.  Clickable comes from the content of the `xref`.  But translation to HTML produces output very similar to the second one.

With Collapsible Math **on**, the second knowl just collapses, while the third (new) one, behaves properly.  First has always worked properly for me.  This is just on Firefox, Chrome is fine.

#1039 continues to solve the problem, so I suggest we just wait.

**Comment by rbeezer** [comment]
#1039 **does not solve** this problem.   The root culprit is the knowl with "Fundamental Theorem of Calculus" as the clickable.  It is "long enough" (heurestically) to be a candidate for collapsing, which it does.

**Comment by davidfarmer** [comment]
Is there a way to make something as "do not collapse this".

**Comment by davidfarmer** [comment]
This fits the subject of this issue, but maybe not the previosu contents

The knowls in the display math following this paragraph are not working for me:
https://pretextbook.org/examples/sample-article/html/section-7.html#p-303

**Comment by rbeezer** [comment]
Working for me on Android tablet, via Chrome.  But with lots of vertical scrollbars.  I'll  try other browsers in the AM.

**Comment by Alex-Jordan** [comment]
Works for me when MathJax renderer is set to CHTML. But not when the renderer is set to SVG.

**Comment by rbeezer** [comment]
Closing as stale. The `\knowl{}`-in-display-math feature still exists, but this describes a specific 2018 rendering glitch in Firefox tied to the MathJax knowl extension — an upstream behavior. MathJax has since moved to version 4, so the original symptom is unlikely to reproduce as described; please reopen with a current example if a knowl in an `md` still misbehaves.

*Claude Fable 5, acting as a review assistant for Rob Beezer*
